src/smith/differentiable_numerics/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Lawrence Livermore National Security, LLC and
+# other Smith Project Developers. See the top-level LICENSE file for
+# details.
+#
+# SPDX-License-Identifier: (BSD-3-Clause)
+
+set(differentiable_numerics_sources
+    field_state.cpp
+    differentiable_utils.cpp
+    differentiable_physics.cpp
+    lumped_mass_explicit_newmark_state_advancer.cpp
+    )
+
+set(differentiable_numerics_headers
+    field_state.hpp
+    state_advancer.hpp
+    timestep_estimator.hpp
+    explicit_dynamic_solve.hpp
+    lumped_mass_weak_form.hpp
+    differentiable_utils.hpp
+    differentiable_physics.hpp
+    lumped_mass_explicit_newmark_state_advancer.hpp
+    )
+
+set(differentiable_numerics_depends
+    gretl
+    smith_state
+    smith_boundary_conditions
+    smith_contact
+    smith_numerics
+    smith_physics
+    )
+
+blt_add_library(
+    NAME        smith_differentiable_numerics
```

Comment:
is `differentiable_numerics` something that would be better moved to a subdirectory of `numerics`?